openfisca_core/entities/_core_entity.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from openfisca_core.types import TaxBenefitSystem, Variable
+from typing import Any
+
+import os
+
+from .role import Role
+from .typing import Entity
+
+
+class _CoreEntity:
+    """Base class to build entities from."""
+
+    #: A key to identify the entity.
+    key: str
+
+    #: The ``key``, pluralised.
+    plural: str | None
+
+    #: A summary description.
+    label: str | None
+
+    #: A full description.
+    doc: str | None
+
+    #: Whether the entity is a person or not.
+    is_person: bool
```

Comment:
May be we should also change this to something `is_single|is_singular|is_atomic` ?

Reply:
I agree, but that change would make this PR a breaking change. Would it be ok to leave that for another one?

Reply:
Which word would you suggest ?

Reply:
The choice of creating an _abstract_ class __CoreEntity_ that is private is to pave the way to that change.